processor-api/src/main/java/com/synopsys/integration/alert/processor/api/NotificationPreProcessor.java

```diff
@@ -0,0 +1,9 @@
+package com.synopsys.integration.alert.processor.api;
+
+import com.synopsys.integration.alert.common.rest.model.AlertNotificationModel;
+import com.synopsys.integration.alert.processor.api.filter.FilterableNotificationWrapper;
+
+public interface NotificationPreProcessor {
```

Comment:
Any naming suggestions here? I was gonna call it `NotificationWrapper`, but "Wrapper" is already used as a noun for the returned object.

Reply:
This is still an extractor in my opinion so I'd call this one also extractor but use the model it extracts in the name.

Reply:
NotificationExtractor?

Reply:
FilterableNotificationExtractor?